src/communication/NetworkClient.ts

```diff
@@ -219,7 +219,7 @@ export default class NetworkClient {
     });
   }
 
-  async patch<R>(pathname: string, data: {}): Promise<R> {
+  async patch<R>(pathname: string, data: NonNullable<unknown>): Promise<R> {
```

Comment:
This is definitely something we should brainstorm about for a little bit.

The issue is that in TypeScript (at this time), you can't `Exclude` from `unknown` or `any`. `NonNullable<unknown>` is effectively equal to `unknown`, and ‒ surprisingly ‒ `null` is actually an acceptable value for it.

I'm not entirely sure what the most helpful types for `data` in `post` and `patch` and `context` in `delete` would be. I think we're looking for "any object which might have an `idempotencyKey` property" for `post` and `delete`; and "any object" for `patch`. Do you have any thoughts on this?

Reply:
It turns out that [since TypeScript 4.8](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-8.html), `unknown` is "close in spirit" to `{} | undefined | null`. This means `NonNullable<unknown>` is ‒ since 4.8 ‒ equal to `{}`.

However, we've decided to go with `Record<string, unknown>`. Although Axios can send a broader variety of data, this library will only send (string-keyed) objects.